subgraphs/uniswap-v3/src/common/creators.ts

```diff
@@ -441,3 +442,79 @@ export function createSwapHandleVolumeAndFees(
   token0.save();
   token1.save();
 }
+
+export function collectFees(
+  event: ethereum.Event,
+  owner: Address,
+  amount0: BigInt,
+  amount1: BigInt
+): void {
+  // update fee growth
+  let pool = getLiquidityPool(event.address.toHexString());
+  let poolAmounts = getLiquidityPoolAmounts(event.address.toHexString());
+  let protocol = getOrCreateDex();
+
+  // Update the pool balances for collected fees in native amount
+  pool.inputTokenBalances = [
+    pool.inputTokenBalances[INT_ZERO].minus(amount0),
+    pool.inputTokenBalances[INT_ONE].minus(amount1),
+  ];
+
+  let token0 = getOrCreateToken(pool.inputTokens[INT_ZERO]);
+  let token1 = getOrCreateToken(pool.inputTokens[INT_ONE]);
+
+  // Get formatted amounts collected.
+  let amount0Converted = convertTokenToDecimal(amount0, token0.decimals);
+  let amount1Converted = convertTokenToDecimal(amount1, token1.decimals);
+
+  // Get the value in USD of the deposit
+  let amountUSD = amount0Converted
+    .times(token0.lastPriceUSD!)
+    .plus(amount1Converted.times(token1.lastPriceUSD!));
+
+  // Update the pool balances for collected fees in converted amount
+  poolAmounts.inputTokenBalances = [
+    poolAmounts.inputTokenBalances[INT_ZERO].minus(amount0Converted),
+    poolAmounts.inputTokenBalances[INT_ONE].minus(amount1Converted),
+  ];
+
+  // Subtract the TVL of the pool from the protocol before fee collection
+  protocol.totalValueLockedUSD = protocol.totalValueLockedUSD.minus(
+    pool.totalValueLockedUSD
```

Comment:
I freaked out here for a second 😅 

Can you maybe put the addition and substraction closer together? Something like:

```
let oldPoolTVL = pool.totalValueLockedUSD;

// update pool TVL
[...]

protocol.totalValueLockedUSD = protocol.totalValueLockedUSD
    .minus(oldPoolTVL)
    .plus(pool.totalValueLockedUSD)
```

Or alternatively:
```
let delta = pool.totalValueLockedUSD.minus(oldPoolTVL);
protocol.totalValueLockedUSD = protocol.totalValueLockedUSD.plus(delta);
```

Reply:
Sure! I fixed this